Update master

MakisH opened the pull request:
It has been a long time since we updated `master`, but it is now the time to, together with the restructured tutorials.

A lot has happened since the last update:
- We moved the wiki pages into the `docs/` folder here. These pages are now rendered in the [new adapter documentation](https://precice.org/adapter-openfoam-overview.html).
- We moved and all the [tutorials](https://precice.org/tutorials.html) to the preCICE website.
- The fluid-fluid (FF) module is now integrated into the main branch. You can find a new [partitioned pipe tutorial on the website](https://precice.org/tutorials-partitioned-pipe.html). See the changes in https://github.com/precice/openfoam-adapter/pull/67
- We cleaned up a few non-const references to boundary fields. See https://github.com/precice/openfoam-adapter/pull/132
- We added (and further improved) a 2D mode. See https://github.com/precice/openfoam-adapter/pull/147
- We refactored the `Force` class. See https://github.com/precice/openfoam-adapter/pull/148
- We added an assertion to restrict the user in valid orientation of 2D cases. See https://github.com/precice/openfoam-adapter/pull/155
- We further improved the `Allwmake` and `Allclean` scripts, dropping support for preCICE 1.x. See https://github.com/precice/openfoam-adapter/pull/157
- We disabled debug messages by default, to reduce visual noise and logs length. See https://github.com/precice/openfoam-adapter/commit/a942c8dc6a9f9ec29f0bb1d6625501657cdd8b65

@DavidSCN have I forgotten anything else worth mentioning?

DavidSCN reviewed (approved):
No, looks complete, good summary :+1:

MakisH commented:
Thank you! I will also add a CHANGELOG, actually, we have delayed this too long.

MakisH commented:
I also removed the Travis badge from the README, as Travis is not in a valid state at the moment.

I added a `CHANGELOG.md`, a `CONTRIBUTING.md`, as well as an updated issue template and a pull request template.
